### ISSUE TITLE
Crash in tab module

### ISSUE THREAD

**Issue by sorenrade** [zellij-org/zellij#1734]
**Basic information**

`zellij --version`: 0.31.3
`stty size`: 50 196
`uname -av`: Linux fresh 5.15.0-46-generic # 49-Ubuntu SMP Thu Aug 4 18:03:25 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

alacritty 0.11.0-dev (ebc6922e)
helix 22.08.1 (66276ce6)
nushell: 0.68.1

###  Error
```
Error occurred in server:

  × Thread 'screen' panicked.
  ├─▶ Originating Thread(s)
  │   	1. screen_thread: HandlePtyBytes
  │
  ├─▶ At /home/ahe/.cargo/registry/src/github.com-1ecc6299db9ec823/zellij-server-0.31.3/src/tab/mod.rs:1031:75
  ╰─▶ called `Option::unwrap()` on a `None` value
```

### Tail of /tmp/zellij-1001/zellij-log/zellij.log
``` 
... lots of similar warnings ...
WARN   |zellij_server::panes::gri| 2022-09-15 20:53:24.920 [screen    ] [/home/ahe/.cargo/registry/src/github.com-1ecc6299db9ec823/zellij-server-0.31.3/src/panes/grid.rs:2744]: Unhandled esc_dispatch: 92->[] 
WARN   |zellij_server::panes::gri| 2022-09-15 20:53:24.920 [screen    ] [/home/ahe/.cargo/registry/src/github.com-1ecc6299db9ec823/zellij-server-0.31.3/src/panes/grid.rs:2207]: Unhandled osc: [[49, 51, 51], [68], [48]] 
WARN   |zellij_server::panes::gri| 2022-09-15 20:53:24.920 [screen    ] [/home/ahe/.cargo/registry/src/github.com-1ecc6299db9ec823/zellij-server-0.31.3/src/panes/grid.rs:2744]: Unhandled esc_dispatch: 92->[] 
ERROR  |zellij_utils::errors     | 2022-09-15 20:53:28.052 [screen    ] [/home/ahe/.cargo/registry/src/github.com-1ecc6299db9ec823/zellij-utils-0.31.3/src/errors.rs:94]: Panic occured:
             thread: screen
             location: At /home/ahe/.cargo/registry/src/github.com-1ecc6299db9ec823/zellij-server-0.31.3/src/tab/mod.rs:1031:75
             message: called `Option::unwrap()` on a `None` value 
INFO   |zellij_server::wasm_vm   | 2022-09-15 20:53:28.054 [wasm      ] [/home/ahe/.cargo/registry/src/github.com-1ecc6299db9ec823/zellij-server-0.31.3/src/wasm_vm.rs:213]: wasm main thread exits 
ERROR  |zellij_utils::errors     | 2022-09-15 20:53:28.641 [async-std/runti] [/home/ahe/.cargo/registry/src/github.com-1ecc6299db9ec823/zellij-utils-0.31.3/src/errors.rs:94]: Panic occured:
             thread: async-std/runtime
             location: At /home/ahe/.cargo/registry/src/github.com-1ecc6299db9ec823/zellij-server-0.31.3/src/terminal_bytes.rs:122:14
             message: called `Result::unwrap()` on an `Err` value: "SendError(..)" 

```

**Further information**
This is the first time this has happened. I had a long scrollback from nushell in my left pane. I entered search mode and then hit e to open in my editor (helix). If I remember correctly it rendered the editor for a second before crashing.

Hope this helps, thanks!

**Comment by har7an** [comment]
Thanks for reporting this issue!

That clearly shouldn't have happened. Unfortunately the log you posted on its own doesn't help narrowing the problem down a lot (which is our fault, not yours). The error indicates that zellij couldn't find a terminal pane which it assumed to exist. Is this a reproducible issue?

For anyone wanting to pick this up: The panic occurs in [`Tab::write_to_pane_id`][1]. I assume the chain of function calls that leads to this is (entirely speculative at this point):

1. `screen.rs: screen_thread_main()`
    - In the loop upon event `ScreenInstruction::PtyBytes`
    - Which calls `tab.handle_pty_bytes()`
5. `tab/mod.rs: Tab::handle_pty_bytes`
6. `tab/mod.rs: Tab::process_pty_bytes`
7. `tab/mod.rs: Tab::write_to_pane_id`


[1]: https://github.com/zellij-org/zellij/blob/588167f38e9581a9d6228577829fe4d6b15c34dc/zellij-server/src/tab/mod.rs#L1031

**Comment by rademonster** [comment]
Haven't hit it again unfortunately. Feel free to close this and I'll reopen if I find a way to reproduce. Thanks.

**Comment by har7an** [comment]
That's okay. I'll leave the issue open until we implement better error reporting around the portion of the code where your bug originated. Hopefully by the time you hit the bug again (in another release) it will tell us more about itself.

**Comment by imsnif** [comment]
@har7an - I'm pretty sure this happens because of a race condition when opening the in-place editor.

I'm guessing that what happens is that a few extra bytes come from the pane we're editing while the editor is open and then we can't find it. The reason we can't find it is that it's in `suppressed_panes` and we're only looking in `tiled_panes` and `floating_panes`. The fix for this is 99% to also look in `suppressed_panes` before that unwrap.

**Comment by har7an** [comment]
@imsnif Guess so, I started adding error handling to the code in `tab` yesterday and found a very similar code snippet in `process_pty_bytes` - which does query `suppressed_panes`. Shall I fix it or are you already at it?

**Comment by imsnif** [comment]
If you can do it, that'll be great. I'm very much in the midst of other stuff :)
You can probably test this manually by running a loop in a pane that throws some output to the screen and then opening the editor.

**Comment by har7an** [comment]
Will be fixed in #1749, also added a test to prevent regressions in the future. I won't be near a PC when the CI finishes, so I'll merge it tomorrow. Thanks again for reporting @sorenrade and thanks @imsnif for the hint!
